File: C07_WHSMail/WHSMailVB/WHSMailCommon/Contracts/IWHSMailService.vb
Imports Microsoft.VisualBasic
Imports System.Collections.Generic
Imports System.ServiceModel
Imports WHSMailCommon.Entities

Namespace WHSMailCommon.Contracts
	' list of methods of the WHSMailService service
	<ServiceContract()> _
	Public Interface IWHSMailService
		<OperationContract> _
		Function GetFolders() As List(Of Folder)

		<OperationContract> _
		Function GetMessages(ByVal entryID As String, ByVal numPerPage As Integer, ByVal pageNum As Integer) As List(Of Email)

		<OperationContract> _
		Function GetMessage(ByVal entryID As String) As Email
	End Interface
End Namespace

File: C07_WHSMail/WHSMailCS/WHSMailCommon/Contracts/IWHSMailService.cs
using System.Collections.Generic;
using System.ServiceModel;
using WHSMailCommon.Entities;

namespace WHSMailCommon.Contracts
{
	// list of methods of the WHSMailService service
	[ServiceContract()]
	public interface IWHSMailService
	{
		[OperationContract]
		List<Folder> GetFolders();

		[OperationContract]
		List<Email> GetMessages(string entryID, int numPerPage, int pageNum);

		[OperationContract]
		Email GetMessage(string entryID);
	}
}
